simplesat/sat/policy/undetermined_clause_policy.py

```diff
@@ -89,13 +93,12 @@ def get_next_package_id(self, assignments, clauses):
         assert assignments.get(candidate_id) is None, \
             "Trying to assign to a variable which is already assigned."
 
-        if not self.prefer_installed:
-            # If this exact package version is available locally, use it.
-            # NOTE: when repository priority is implemented, this will cause
-            # the virtual <installed> repository to act as if it always has the
-            # highest priority.
-            key = self._package_key(candidate_id)
-            candidate_id = self._preferred_package_ids.get(key, candidate_id)
+        # If this exact package version is available locally, use it.
+        # NOTE: when repository priority is implemented, this will cause
+        # the virtual <installed> repository to act as if it always has the
+        # highest priority.
+        key = self._package_key(candidate_id)
```

Comment:
If the block is removed, you can end up removing the installed package and then re-installing same version from the remote.
It may be best to remove this block and add some post-processing to ensure that we use the installed versions if possible.